CIP-data-compression/README.md

```diff
@@ -0,0 +1,273 @@
+
+An attacker can still send malicious data to the Cardano Node. However, in such cases, the worker process will immediately crash, and the caller process will receive a corresponding notification.
+
+Given the high quality of the ZStandard library, such a crash is likely indicative of an attack attempt, warranting an appropriate response:
+- **Block the malicious peer** to prevent further communication and mitigate potential Denial-of-Service attacks.
+- **Restart the worker process** to continue handling requests from a safe state.
+
+The provided implementation is compact—approximately **200 lines of code** for [the worker process](./secure-zstd/lib/seczstd/worker.c) and **100 lines of code** for [the caller](./secure-zstd/lib/seczstd/caller.c). This allows for **easy security audits** compared to auditing the full ZStandard decompression library, which consists of about **15,000** lines of code.
+
+Furthermore, the **Isolation via IPC** approach, extended with ```seccomp```, can be similarly applied to other untrusted data-processing tasks, such as newer but less-tested cryptographic libraries (e.g., potential Plutus builtins) and other use cases.
+
+### Indicating Protocol Version Support in Stake Pool Operator On-Chain Data
+
+Some clients may prefer receiving block data in a compressed form to save bandwidth. Providing a quick way to identify stake pool operators that support compressed transfers would be valuable.
+
+Cardano stake pool operators can already publish a URL linking to a publicly available JSON file containing additional information about their pool. These URLs are recorded on-chain and can also be retrieved through a stake pool metadata aggregation service (SMASH).
+
+This proposal extends the metadata JSON by adding an optional key, ```protocolVersions```, which contains a list of currently supported protocol versions as a JSON array.
+
+```
+protocolVersions = [12, 13, 15]
+```
+
+This allows clients to quickly determine whether a stake pool operator supports the latest protocol version, including features such as compressed block transfers.
+This is particularly important during node bootstrapping, when a new node does not yet have an up-to-date list of block-producing nodes but needs to synchronize as quickly as possible.
+
+## Rationale: how does this CIP achieve its goals?
+
+- Analyzes how support for data compression can serve as a technical foundation for drastically improving Cardano's scalability.
+- Evaluates the performance of the ZStandard compression algorithm on Cardano mainnet data.
+- Demonstrates how data compression, combined with more efficient use of affordable, high-performance server hardware, can scale Cardano’s transaction throughput by multiple orders of magnitude.
+- Proposes specific changes to the networking protocol to enable data compression.
+
+## Path to Active
+
+### Possible Support Levels
```

Comment:
To avoid confusion both about this CIP & the general CIP process (since these documents are often copied & used as templates by future authors) we should stick to the standard format by moving these specification alternatives more properly into the _Specification_ section itself... and then referring to that section in the [usual 2 sections of the Path to Active](https://github.com/cardano-foundation/CIPs/blob/master/CIP-0001/README.md#path-to-active).